bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3070,7 +3074,7 @@ void ReplicaImp::goToNextView() {
 bool ReplicaImp::tryToEnterView() {
   ConcordAssert(!currentViewIsActive());
   ConcordAssertEQ(activeExecutions_, 0);
-  std::vector<PrePrepareMsg *> prePreparesForNewView;
+  std::vector<std::shared_ptr<PrePrepareMsg>> prePreparesForNewView;
 
   bool enteredView =
       viewsManager->tryToEnterView(getCurrentView(), lastStableSeqNum, lastExecutedSeqNum, &prePreparesForNewView);
```

Comment:
I think this function should return a vectorm not a bool. If the vector is not empty, it means in entered view, if empty id didnt.
Vector can be moved when return - if you get a warning. means there is an RVO/RVNO and you dont need to move.

Reply:
Done

Reply:
Unfortunatelly there is a rare case when this vector is empty but the view should be new, so it cannot be changed. The test which exposes that is test_view_changes_at_startup